drivers/usb/device/usb_dc_rpi_pico.c

```diff
@@ -0,0 +1,819 @@
+/*
```

Comment:
["Raspberry Pi Pico is a tiny, fast, and versatile board built using RP2040" ](https://www.raspberrypi.com/products/raspberry-pi-pico/), I do not see any reason to postfix drivers with a board abbreviation. You are implementing USB device controller driver for RP2040 SoC, not a board.

Reply:
I agree with @jfischer-no, we should instead rename the UART and GPIO drivers IMHO.

Reply:
@yonsch do you agree?

Reply:
> I would argue for `rp2` suffix in this case,

:+1:

Reply:
The "pico" naming was chosen because the SDK is named "pico-sdk", so we assumed pico is the name of the series, not just the board. The 2 in "rp2" denotes the number of cores, so if the next chip in the series happens to have one core, we will need another refactor. Also, since the drivers are written using pico-sdk, I think it makes sense that the drivers are also denoted with "pico".
I think we should stick with pico, at least until we know more (e.g. another chip is released or an official announcement about future chips).

Reply:
Anyone want to wade into this again? Stick w/ the convention everywhere else?

Reply:
IMHO you can continue with current naming scheme (rpi_pico), this can be changed later within a few minutes. I added dev-review label to discuss it there.